### PR TITLE
resolving adaptors task.

### PR DESCRIPTION
**Pull request by collderart**
Weird situation here with Empty string test. I checked my code in Visual Studio and it passing an empty string.

### CODE REVIEW

File: adaptors task/student/src/impl.cpp [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+#include <stack>
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+bool isValid(const std::string& source){
+    if (source.empty() or source.size()==1)
+    {
+        return false;
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:12]: if source is empty, string is valid :)
only brackets should be checked

File: adaptors task/student/src/impl.cpp [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
+    }
+    std::stack<char> res;
+
+    for (auto stringCh : source) {
+        for (auto reg : brackets) {
+            if (stringCh == reg.first)
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:18]: name this condition

File: adaptors task/student/src/impl.cpp [hunk 3]
```diff
@@ -1,8 +1,35 @@
...
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+bool isValid(const std::string& source){
+    if (source.empty() or source.size()==1)
+    {
+        return false;
+    }
+    std::stack<char> res;
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:14]: don't use abbreviations

File: adaptors task/student/src/impl.cpp [hunk 4]
```diff
@@ -1,8 +1,35 @@
...
+
+    for (auto stringCh : source) {
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:16]: please give a better name

File: adaptors task/student/src/impl.cpp [hunk 5]
```diff
@@ -1,8 +1,35 @@
...
+        for (auto reg : brackets) {
+            if (stringCh == reg.first)
+                {
+                    res.push(stringCh);
+                    continue;
+                }
+                else if (stringCh == reg.second)
+                {
+                    if (res.empty() or source.at(0) == reg.second or res.top() != reg.first)
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:25]: always name complex conditions!

**Review by smay1613 — CHANGES_REQUESTED** [review]

File: adaptors task/student/src/impl.cpp [hunk 6]
```diff
@@ -6,30 +6,31 @@ const std::vector<std::pair<char, char>> brackets {
...
+    for (auto currentChar : source) {
+        for (auto bracket : brackets) {
+            if (currentChar == bracket.first)
+            {
+                checkString.push(currentChar);
+                continue;
+            }
+            else if (currentChar == bracket.second)
+            {
+                if (source.at(0) == bracket.second or checkString.top() != bracket.first)
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp]: why source.at(0)???

File: adaptors task/student/src/impl.cpp [hunk 7]
```diff
@@ -6,30 +6,31 @@ const std::vector<std::pair<char, char>> brackets {
...
-                    if (res.empty() or source.at(0) == reg.second or res.top() != reg.first)
-                    {
-                        return false;
-                    }
-                    res.pop();
-                    continue;
+                    return false;
                 }
+                checkString.pop();
+                continue;
```
Comment (smay1613) [inline adaptors task/student/src/impl.cpp:27]: do you need it?

Comment (collderart) [reply adaptors task/student/src/impl.cpp:27]: Right, I don't need it.

File: adaptors task/student/src/impl.cpp (hunk 6 above)
Comment (collderart) [reply adaptors task/student/src/impl.cpp]: I changed it to checkString.empty() in the last commit

File: adaptors task/student/src/impl.cpp [hunk 8]
```diff
@@ -6,30 +6,31 @@ const std::vector<std::pair<char, char>> brackets {
...
-        for (auto reg : brackets) {
-            if (stringCh == reg.first)
-                {
-                    res.push(stringCh);
-                    continue;
-                }
-                else if (stringCh == reg.second)
+    for (auto currentChar : source) {
+        for (auto bracket : brackets) {
+            if (currentChar == bracket.first)
```
[the remark this answers is not in the document]
Comment (collderart) [reply adaptors task/student/src/impl.cpp:15]: I'm sorry, what do you mean by that?

You mean make new function for complex conditions and just call it in main function?

[the remark this answers is not in the document]
Comment (smay1613) [reply adaptors task/student/src/impl.cpp:15]: Use named const bools

**Pull request closed without merging**